make/Defs.gmk

```diff
@@ -141,7 +141,15 @@ CP 	= /bin/cp
 DIFF 	= /usr/bin/diff
 ECHO	= /bin/echo
 FIND	= /usr/bin/find
+ifeq ($(SYSTEM_UNAME), FreeBSD)
+GREP  := $(shell if [ -r /usr/local/bin/ggrep ]; then \
```

Comment:
Maybe consider using ``which ggrep`` instead?

Reply:
Wouldn't this pretty much just be the same as just not specifying the path at all, and let the shell find it in $PATH? Which was [one of the complaints](https://github.com/openjdk/jtreg/pull/237#issuecomment-2543058251) against my first patch.

Reply:
Okay, people have apparently varying opinions on this. :) Don't want to put you in a "damned if you do, damned if you don't" situation, so just ignore my comment and go with your solution.

Reply:
I don't have any strong opinions about this, but for FreeBSD's part it would actually be slightly better to let the shell find the binaries in $PATH. But it's not important for my progress now, so I figured I'd deal with it if it comes up :)

Reply:
Maybe the concern is becoming obsolete, but the general design focus of this and similar scripts was to lock down everything, and avoid any variation that might occur as a result of individuals setting non-standard values for PATH and similar path-related variables -- CLASSPATH anyone?

But, a reasonable compromise would be to limit the use of `which` to this file (`Defs.gmk`) and maybe to within suitable `ifeq ($(SYSTEM_UNAME), FreeBSD)` constructs.

That all being said, I would defer to the expertise and style-advice of folk like @magicus on the Build team,  but with some amount of caveat for overall style consistency, as much as is reasonable.

Reply:
If the path is hardcoded as e.g. `/usr/bin/foo`, I tend to agree with you. That means that an official system tool is needed and confirmed working. However, hard-coding `/usr/local/bin/foo` is another matter altogether. It means there is no standard system tool that is working, and we presume the user has installed a "correct" tool locally.

Reply:
If the path is hardcoded as e.g. `/usr/bin/foo`, I tend to agree with you. That means that an official system tool is needed and confirmed working. However, hard-coding `/usr/local/bin/foo` is another matter altogether. It means there is no standard system tool that is working, and we presume the user has installed a "correct" tool locally.

Reply:
(uness BSD does thing different in this regard and install all user-installed packages in /usr/local?)

Reply:
> (uness BSD does thing different in this regard and install all user-installed packages in /usr/local?)

Yep, that's how FreeBSD does things. (Technically the prefix can be something different than /usr/local, but that's the default.)

Reply:
Ok, in that case this makes totally sense.